modules/features2d/test/test_lshindex_flannbased_matcher.cpp

```diff
@@ -0,0 +1,534 @@
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *************************************************************************/
+
+/*
+ For OpenCV2.4/OpenCV3.0
+
+ Test for Pull Request # 3829
+ https://github.com/Itseez/opencv/pull/3829
+
+ This test code creates brute force matcher for accuracy of reference, and the test target matcher.
+ Then, add() and train() transformed query image descriptors, and some outlier images descriptors to both matchers.
+ Then, compared with the query image by match() and findHomography() to detect outlier and calculate accuracy.
+ And each drawMatches() images are saved, if SAVE_DRAW_MATCHES_IMAGES is true.
+ Finally, compare accuracies between the brute force matcher and the test target matcher.
+
+ The lsh algorithm uses std::random_shuffle in lsh_index.h to make the random indexes table.
+ So, in relation to default random seed value of the execution environment or by using "srand(time(0)) function",
+ the match time and accuracy of the match results are different, each time the code ran.
+ And the match time becomes late in relation to the number of the hash collision times.
+*/
+
+#include "test_precomp.hpp"
+#include "opencv2/ts.hpp"    // for FilePath::CreateFolder()
+#include <time.h> // for time()
+
+// If defined, the match time and accuracy of the match results are a little different, each time the code ran.
+//#define INIT_RANDOM_SEED
+
+// If defined, some outlier images descriptors add() the matcher.
+#define TRAIN_WITH_OUTLIER_IMAGES
+
+// If true, save drawMatches() images.
+#define SAVE_DRAW_MATCHES_IMAGES false
+
+// if true, verbose output
+#define SHOW_DEBUG_LOG true
+
+#if CV_MAJOR_VERSION==2
+#define OrbCreate new ORB(4000)
+#elif CV_MAJOR_VERSION==3
+#define OrbCreate ORB::create(4000)
+#define AKazeCreate AKAZE::create()
+#endif
+
+using namespace cv;
```

Comment:
instead of "using namespace cv" put all the tests into cv or cv::test namespace.